third_party/ruby/ruby-2.4.BUILD

```diff
@@ -729,75 +758,203 @@ cc_library(
+            "-DHAVE_SOCKETPAIR",
+            "-DHAVE_GETHOSTNAME",
+            "-DENABLE_IPV6",
+            "-DHAVE_CONST_AF_UNIX",
+            "-DHAVE_CONST_SCM_RIGHTS",
+            "-DFD_PASSING_WORK_WITH_RECVMSG_MSG_PEEK",
+            "-DHAVE_GETNAMEINFO",
+            "-DHAVE_GETADDRINFO",
+            "-Wno-dangling-else",
+            "-Wno-unused-const-variable",
+        ],
+        ":darwin": [
+            "-DHAVE_SYS_UIO_H",
+            "-DHAVE_NETINET_IN_SYSTM_H",
+            "-DHAVE_NETINET_TCP_H",
+            "-DHAVE_NETINET_TCP_FSM_H",
+            "-DHAVE_NETINET_UDP_H",
+            "-DHAVE_ARPA_INET_H",
+            "-DHAVE_NET_ETHERNET_H",
+            "-DHAVE_SYS_UN_H",
+            "-DHAVE_IFADDRS_H",
+            "-DHAVE_SYS_IOCTL_H",
+            "-DHAVE_SYS_SOCKIO_H",
+            "-DHAVE_NET_IF_H",
+            "-DHAVE_SYS_PARAM_H",
+            "-DHAVE_SYS_UCRED_H",
+            "-DHAVE_NET_IF_DL_H",
+            "-DHAVE_ARPA_NAMESER_H",
+            "-DHAVE_RESOLV_H",
+            "-DHAVE_STRUCT_SOCKADDR_SA_LEN",
+            "-DHAVE_ST_SA_LEN",
+            "-DHAVE_STRUCT_SOCKADDR_IN_SIN_LEN",
+            "-DHAVE_ST_SIN_LEN",
+            "-DHAVE_STRUCT_SOCKADDR_IN6_SIN6_LEN",
+            "-DHAVE_ST_SIN6_LEN",
+            "-DHAVE_TYPE_STRUCT_SOCKADDR_UN",
+            "-DHAVE_STRUCT_SOCKADDR_UN_SUN_LEN",
+            "-DHAVE_ST_SUN_LEN",
+            "-DHAVE_TYPE_STRUCT_SOCKADDR_DL",
+            "-DHAVE_TYPE_STRUCT_SOCKADDR_STORAGE",
+            "-DHAVE_TYPE_STRUCT_ADDRINFO",
+            "-DHAVE_TYPE_SOCKLEN_T",
+            "-DHAVE_TYPE_STRUCT_IN_PKTINFO",
+            "-DHAVE_STRUCT_IN_PKTINFO_IPI_SPEC_DST",
+            "-DHAVE_ST_IPI_SPEC_DST",
+            "-DHAVE_TYPE_STRUCT_IN6_PKTINFO",
+            "-DHAVE_TYPE_STRUCT_IP_MREQ",
+            "-DHAVE_TYPE_STRUCT_IP_MREQN",
+            "-DHAVE_TYPE_STRUCT_IPV6_MREQ",
+            "-DHAVE_STRUCT_MSGHDR_MSG_CONTROL",
+            "-DHAVE_ST_MSG_CONTROL",
+            "-DHAVE_SOCKET",
+            "-DHAVE_SENDMSG",
+            "-DHAVE_RECVMSG",
+            "-DHAVE_FREEHOSTENT",
+            "-DHAVE_FREEADDRINFO",
+            "-DHAVE_GAI_STRERROR",
+            "-DGAI_STRERROR_CONST",
+            "-DHAVE_INET_NTOP",
+            "-DHAVE_INET_PTON",
+            "-DHAVE_GETSERVBYPORT",
+            "-DHAVE_GETIFADDRS",
+            "-DHAVE_GETPEEREID",
+            "-DHAVE_IF_INDEXTONAME",
+            "-DNEED_IF_INDEXTONAME_DECL",
+            "-DHAVE_IF_NAMETOINDEX",
+            "-DNEED_IF_NAMETOINDEX_DECL",
+            "-DHAVE_GETIPNODEBYNAME",
+            "-DHAVE_GETHOSTBYNAME2",
+            "-DHAVE_SOCKETPAIR",
+            "-DHAVE_GETHOSTNAME",
+            "-DENABLE_IPV6",
+            "-DINET6",
+            "-DHAVE_CONST_AF_UNIX",
+            "-DHAVE_CONST_SCM_RIGHTS",
+            "-DHAVE_GETNAMEINFO",
+            "-DHAVE_GETADDRINFO",
+            "-Wno-dangling-else",
+            "-Wno-unused-const-variable",
+        ]
+    }),
```

Comment:
There's a lot of duplication between the two branches of the `select` here, would it be quick to pull out the common options into one list, and concatenate with a `select` of the unique parts?

Reply:
It's possible: the reason I didn't do that is because this was directly copied from the output of a Ruby configure step, and I thought that treating this as output directly from a script (instead of trying to process it) might help us in the future—e.g. if the configuration changes across Ruby versions, then all we need to do is drop in a different set of flags instead of having to re-comb through the more human-readable-but-now-modified version.

Reply:
That's a good point. I don't have a good sense right now for how difficult it's going to be to maintain this through different versions of ruby. Maybe doing what we can to ease that transition now is the best path forward?